sst/Assemble.cc

```diff
@@ -0,0 +1,186 @@
+  ASSERT(instrInfo != nullptr, "Failed to create LLVM instruction info");
+
+  // Create MC asm backend
+  std::unique_ptr<llvm::MCAsmBackend> asmBackend(
+      target->createMCAsmBackend(*subtargetInfo, *regInfo, options));
+  ASSERT(asmBackend != nullptr, "Failed to create LLVM asm backend");
+
+  // Create MC code emitter
+  std::unique_ptr<llvm::MCCodeEmitter> codeEmitter(
+      target->createMCCodeEmitter(*instrInfo, *regInfo, context));
+  ASSERT(codeEmitter != nullptr, "Failed to create LLVM code emitter");
+
+  // Create MC object writer
+  llvm::SmallVector<char, 1024> objectStreamData;
+  llvm::raw_svector_ostream objectStream(objectStreamData);
+  std::unique_ptr<llvm::MCObjectWriter> objectWriter =
+      asmBackend->createObjectWriter(objectStream);
+  ASSERT(objectWriter != nullptr, "Failed to create LLVM object writer");
+
+  // Create MC object streamer
+  std::unique_ptr<llvm::MCStreamer> objectStreamer(
+      target->createMCObjectStreamer(
+          llvm::Triple(triple), context, std::move(asmBackend),
+          std::move(objectWriter), std::move(codeEmitter), *subtargetInfo,
+          options.MCRelaxAll, options.MCIncrementalLinkerCompatible, false));
+  ASSERT(objectStreamer != nullptr, "Failed to create LLVM object streamer");
+
+  // Create MC asm parser
+  std::unique_ptr<llvm::MCAsmParser> asmParser(
+      llvm::createMCAsmParser(srcMgr, context, *objectStreamer, *asmInfo));
+  ASSERT(asmParser != nullptr, "Failed to create LLVM asm parser");
+
+  // Create MC target asm parser
+  std::unique_ptr<llvm::MCTargetAsmParser> targetAsmParser(
+      target->createMCAsmParser(*subtargetInfo, *asmParser, *instrInfo,
+                                options));
+  ASSERT(asmParser != nullptr, "Failed to create LLVM target asm parser");
+  asmParser->setTargetParser(*targetAsmParser);
+
+  // Run asm parser to generate assembled object code
+  ASSERT(!asmParser->Run(false), "");
+
+  // Create ELF object from output
+  llvm::StringRef objectData = objectStream.str();
+  auto elfOrErr = llvm::object::ELFFile<
+      llvm::object::ELFType<llvm::support::little, true>>::create(objectData);
+  ASSERT(!elfOrErr.takeError(), "Failed to load ELF object");
+  auto& elf = *elfOrErr;
+
+  // Get handle to .text section
+  auto textOrErr = elf.getSection(2);
+  ASSERT(!textOrErr.takeError(), "Failed to find .text section");
+  auto& text = *textOrErr;
+
+  // Get reference to .text section data
+#if SIMENG_LLVM_VERSION < 12
+  auto textDataOrErr = elf.getSectionContents(text);
+#else
+  auto textDataOrErr = elf.getSectionContents(*text);
+#endif
+  ASSERT(!textDataOrErr.takeError(), "Failed to get .text contents");
+  llvm::ArrayRef<uint8_t> textData = *textDataOrErr;
+
+  // Make copy of .text section data
+  codeSize_ = textData.size();
+  code_ = new uint8_t[codeSize_];
+  std::copy(textData.begin(), textData.end(), code_);
+}
+#else
+void Assembler::assemble(const char* source, const char* triple) {}
```

Comment:
Got Idea! I have changed it a bit though instead of assert I've thrown a descriptive error. This further eliminates the need for conditional compilation in SimEngCoreWrapper

Reply:
Whatever works. Solution for output looks good and further elimination of an ```ifdef``` is great